dandischema/metadata.py

```diff
@@ -424,22 +418,17 @@ def migrate(
         raise ValueError(msg)
     # ----------------------------------------------------------------
 
-    # Ensure the target DANDI schema version is at least the DANDI schema version
-    # of the provided instance
-    if obj_ver_tuple > target_ver_tuple:
-        raise ValueError(f"Cannot migrate from {obj_ver} to lower {to_version}.")
-
     # Optionally validate the instance against the DANDI schema it specifies
     # before migration
     if not skip_validation:
-        _validate_obj_json(obj, _get_jsonschema_validator(obj_ver, "Dandiset"))
+        _validate_obj_json(obj, _get_jsonschema_validator(obj_version, "Dandiset"))
```

Comment:
This migration function seems to expect the metadata instance to be a `Dandiset` instance.

Reply:
Validating `PublishedDandiset` instance may create a problem if https://github.com/dandi/dandi-schema/pull/266 is put in. However, I guess we can have it specifically validate against the `PublishDandiset` schema if a `PublishedDandiset` instance is provided.